### PR TITLE
feat(cms-base): regulation price displaying

### PR DESCRIPTION
**Pull request by mkucmus**
closes: #94 

- [x] types enhancement
- [x] cms-base package adjusted
- [x] vue-demo-store adjusted
- [x] tier prices condition to avoid wrong price displaying
- [x] added missing tier prices on pdp built in cms-page 
- [x] check for other occurrences when regulation price can be displayed 
- [x] report found issues in the core

### CODE REVIEW

File: packages/cms-base/components/SwListingProductPrice.vue [hunk 1]
```diff
@@ -9,6 +9,8 @@ const { product } = toRefs(props);
 
 const { price, unitPrice, displayFromVariants, displayFrom, isListPrice } =
   useProductPrice(product);
+const regulationPrice = computed(() => price?.value?.regulationPrice?.price)
```
Comment (patzick) [inline packages/cms-base/components/SwListingProductPrice.vue:12]:
```suggestion
const regulationPrice = computed(() => price.value?.regulationPrice?.price)
```
one too many :)

File: packages/cms-base/components/public/cms/element/CmsElementBuyBox.vue [hunk 2]
```diff
@@ -22,19 +18,10 @@ const { product, changeVariant } = useProduct(
...
-    );
-  } else {
-    return null;
-  }
-});
-const listPrice = computed(() =>
-  getProductCalculatedListingPrice(product.value)
-);
+const { unitPrice, price, tierPrices, isListPrice } = useProductPrice(product);
+const regulationPrice = computed(() => price?.value?.regulationPrice?.price)
```
Comment (patzick) [inline packages/cms-base/components/public/cms/element/CmsElementBuyBox.vue:22]:
```suggestion
const regulationPrice = computed(() => price.value?.regulationPrice?.price)
```

File: packages/types/shopware-6-client/models/checkout/cart/price/CalculatedPrice.d.ts [hunk 3]
```diff
@@ -19,4 +19,8 @@ export type CalculatedPrice = {
     percentage: number;
     apiAlias: string;
   };
+  regulationPrice: {
```
Comment (patzick) [inline packages/types/shopware-6-client/models/checkout/cart/price/CalculatedPrice.d.ts:22]: Does it mean it's always visible if calculated prices are visible?

File: packages/types/shopware-6-client/models/framework/pricing/Price.d.ts [hunk 4]
```diff
@@ -1,6 +1,10 @@
 export type Price = {
-  currencyId: number;
   net: number;
   gross: number;
   linked: boolean;
+  listPrice: null;
+  currencyId: number;
+  regulationPrice: Partial<Price>;
```
Comment (patzick) [inline packages/types/shopware-6-client/models/framework/pricing/Price.d.ts:7]: not the same as in `CalculatedPrice`?

**Review by patzick — CHANGES_REQUESTED** [review]
Looks good, I have some questions for typing :)

if the regular price is always if calculatedPrice exist we could improve:

```ts
const regulationPrice = computed(() => price.value?.regulationPrice.price)
```

or even we can use it without computed directly in the template as it's not something which is reactive
```html
<div class="text-xs flex text-gray-500" v-if="price?.regulationPrice">
    Previously <SharedPrice class="ml-1" :value="price.regulationPrice.price"/>
</div>
```

File: packages/types/shopware-6-client/models/framework/pricing/Price.d.ts (hunk 4 above)
Comment (mkucmus) [reply packages/types/shopware-6-client/models/framework/pricing/Price.d.ts:7]: It's a different place and format :)

**Review by patzick — APPROVED** [review]

**Pull request merged**